plugin-server/src/worker/plugins/run.ts

```diff
@@ -351,17 +356,59 @@ export async function runPluginTask(
     return response
 }
 
-async function getPluginMethodsForTeam<M extends keyof VMMethods>(
+async function getPluginMethodsForTeam<M extends keyof VMMethodsConcrete>(
     hub: Hub,
     teamId: number,
     method: M
-): Promise<[PluginConfig, VMMethods[M]][]> {
+): Promise<[PluginConfig, VMMethodsConcrete[M]][]> {
     const pluginConfigs = hub.pluginConfigsPerTeam.get(teamId) || []
     if (pluginConfigs.length === 0) {
         return []
     }
+
     const methodsObtained = await Promise.all(
         pluginConfigs.map(async (pluginConfig) => [pluginConfig, await pluginConfig?.vm?.getVmMethod(method)])
     )
-    return methodsObtained as [PluginConfig, VMMethods[M]][]
+
+    const methodsObtainedFiltered = methodsObtained.filter(([_, method]) => !!method) as [
+        PluginConfig,
+        VMMethodsConcrete[M]
```

Comment:
This is a nice improvement to skip the duplicate filtering and simplify a lot of the other logic by making the selected method "concrete".